src/main/java/useCases/generator/PropertyGenerator.java

```diff
@@ -1,76 +1,52 @@
 package useCases.generator;
 
+import datagateways.PropertyDataManager;
 import entity.Property;
 
-import java.io.BufferedReader;
-import java.io.FileNotFoundException;
-import java.io.FileReader;
-import java.io.IOException;
 import java.util.ArrayList;
 import java.util.Arrays;
 
 /**
- * The PropertyGenerator class represents an use case responsible for generating property blocks.
+ * The PropertyGenerator class represents a use case responsible for generating property blocks.
  * It reads property information from a CSV file and creates a list of properties based on the data.
  */
 public class PropertyGenerator{
 
     /**
-     * The delimiter used to separate values in the CSV file.
-     */
-    static final String DELIMITER = ",(?=([^\"]*\"[^\"]*\")*[^\"]*$)";
-
-    /**
-     * Creates properties based on the information in the specified CSV file.
+     * Generates a list of properties based on information from a CSV file.
      *
      * @param fileName The name of the CSV file containing property information.
      * @return An ArrayList of Property objects generated from the CSV data.
+     * @throws NumberFormatException    If action values in the CSV file are not in the expected numeric format.
      * @throws IllegalArgumentException If there are missing values in the CSV line.
-     * @throws NumberFormatException    If action values are not in the expected numeric format in the CSV file.
      */
     public static ArrayList<Property> generateProperties(String fileName){
-
-        ArrayList<entity.Property> propertyList = new ArrayList<>();
-
-        try (BufferedReader br = new BufferedReader(new FileReader(fileName))){
-            String line;
-            ArrayList<Integer> priceList = new ArrayList<>(Arrays.asList(0, 0, 0, 0, 0, 0));
-            ArrayList<Integer> taxList = new ArrayList<>(Arrays.asList(0, 0, 0, 0, 0, 0));
-
-            while ((line = br.readLine()) != null) {
-                String[] values = line.split(DELIMITER);
-
-                if (values.length < 12) {
-                    // Handle the case where there are missing values in the CSV line
-                    throw new IllegalArgumentException("Invalid data in the CSV file: " + line);
+        PropertyDataManager dataManager = new PropertyDataManager();
+        ArrayList<String[]> data = dataManager.readData(fileName);
+
+        ArrayList<Property> propertyList = new ArrayList<>();
+        ArrayList<Integer> priceList = new ArrayList<>(Arrays.asList(0, 0, 0, 0, 0, 0));
```

Comment:
Same here, maybe we want to document what each field represents here, so that its easier for people to follow up. 
Or, we can also add the documentation beside the CSVs or something